src/main/java/org/takes/facets/hamcrest/HmRsHeader.java

```diff
@@ -43,78 +46,180 @@
  * @author Eugene Kondrashev (eugene.kondrashev@gmail.com)
  * @version $Id$
  * @since 0.23.3
- * @todo #260:30min Implement additional constructors.
- *  According to #260 there should be also available such constructors
- *  public HmRsHeader(final Matcher<? extends Map.Entry<String,String>> mtchr);
- *  public HmRsHeader(final String header,
- *  final Matcher<? extends Iterable<String>> mtchr);
- *  public HmRsHeader(final String header,
- *  final Matcher<? extends String> mtchr);
- *  public HmRsHeader(final String header, final String value);
  */
 public final class HmRsHeader extends TypeSafeMatcher<Response> {
 
     /**
-     * Expected request header matcher.
+     * Expected response header matcher.
      */
-    private final transient Matcher<? extends Map<? extends CharSequence,
-        ? extends CharSequence>> matcher;
+    private final transient HeaderMatcher matcher;
 
     /**
-     * Expected matcher.
-     * @param mtchr Is expected header matcher.
+     * Ctor.
+     * @param mtchr Matcher
      */
-    public HmRsHeader(final Matcher<? extends Map<? extends CharSequence,
-        ? extends CharSequence>> mtchr) {
+    public HmRsHeader(
+        final Matcher<? extends Map.Entry<String, String>> mtchr) {
         super();
-        this.matcher = mtchr;
+        this.matcher = new EntryHeaderMatcher(mtchr);
     }
 
     /**
-     * Fail description.
-     * @param description Fail result description.
+     * Ctor.
+     * @param header Header name
+     * @param mtchr Matcher
      */
+    public HmRsHeader(final String header,
+        final Matcher<? extends Iterable<String>> mtchr) {
+        super();
+        this.matcher = new IterableHeaderMatcher(header, mtchr);
+    }
+
+    /**
+     * Ctor.
+     * @param header Header name
+     * @param value Header value
+     */
+    public HmRsHeader(final String header, final String value) {
+        this(new EntryMatcher<String, String>(header, value));
+    }
+
     @Override
     public void describeTo(final Description description) {
         this.matcher.describeTo(description);
     }
 
-    /**
-     * Type safe matcher.
-     * @param item Is tested element
-     * @return True when expected type matched.
-     */
     @Override
     public boolean matchesSafely(final Response item) {
         try {
             final Iterator<String> headers = item.head().iterator();
-            headers.next();
+            if (headers.hasNext()) {
```

Comment:
@pinaf Good remark, but I believe I actually fixed the previous behaviour which would throw an exception if there were no headers at all. From the standpoint of the `HmRsHeader`, that means it that it matches against an empty headers collection which should not lead to the exception. Please correct me if you still think otherwise.